collector/internal/tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-lambda/collector/internal/tools
 
-go 1.19
+go 1.23
+
+toolchain go1.23.2
```

Comment:
This PR now has too many changes.  Please revert everything else.

Reply:
dropped all irrelevant changes and rebased on top of latest main.